### PR TITLE
refactor: create helper method for limiter functions

### PR DESCRIPTION
**Pull request by vindard**
## Description

- Standardise the way that we consume limiters
- Add better error reporting granularity for limiters consumed via a single config

### CODE REVIEW

File: src/domain/rate-limit/index.ts
Comment (nicolasburtey) [inline src/domain/rate-limit/index.ts:61]: I have renamed this to `invoiceCreateForSelf` in this pending draft PR: https://github.com/GaloyMoney/galoy/pull/936

**Review by nicolasburtey — APPROVED** [review]

File: src/domain/rate-limit/index.ts
Comment (vindard) [reply src/domain/rate-limit/index.ts:61]: Should I wait until #936 is reviewed/merged to then merge this one?

**Review by dolcalmi — APPROVED** [review]
looks good

File: src/domain/rate-limit/index.ts
Comment (nicolasburtey) [reply src/domain/rate-limit/index.ts:61]: no you can merge it. it's more FI. #936 has some dependancy on #921 also.

**Pull request merged**
